bip32/src/extended_key/public_key.rs

```diff
@@ -135,8 +135,83 @@ where
                 public_key: PublicKey::from_bytes(extended_key.key_bytes)?,
                 attrs: extended_key.attrs.clone(),
             })
+        } else if extended_key.prefix.is_private() {
```

Comment:
You'll need to gate this (as well as the `XPrv` import) on the `secp256k1` feature being available:

```suggestion
        } else if cfg!(feature = "k256") && extended_key.prefix.is_private() {
```

Reply:
How can I feature-gate the `XPrv` import? If `use crate::XPrv` is feature-gated, then this code will not compile when `--no-default-features` because `XPrv` is still referenced on the line below.

Reply:
You can either do:

```rust
#[cfg(feature = "k256")]
use crate::XPrv;
```

or refer to it directly as `crate::XPrv`